toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1332,9 +1402,25 @@ func (g *PkgGraph) fixIntraSpecCycle(trimmedCycle []*PkgNode) (err error) {
 	return
 }
 
+func (g *PkgGraph) replaceCurrentNodeWithPrebuiltNode(currentNode, preBuiltNode, previousNode *PkgNode) {
+	parentNodes := g.To(currentNode.ID())
+	for parentNodes.Next() {
+		parentNode := parentNodes.Node().(*PkgNode)
+		if parentNode.Type == TypeLocalBuild && parentNode.SrpmPath == previousNode.SrpmPath {
+			g.RemoveEdge(parentNode.ID(), currentNode.ID())
+
+			err := g.AddEdge(parentNode, preBuiltNode)
+			if err != nil {
+				logger.Log.Errorf("Adding edge failed for %v -> %v", parentNode, preBuiltNode)
+				return
+			}
+		}
+	}
+}
+
 // fixPrebuiltSRPMsCycle attempts to fix a cycle if at least one node is a pre-built SRPM.
 // If a cycle can be fixed, edges representing the build dependencies of the pre-built SRPM will be removed.
-func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
+func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode, resolveCyclesFromUpstream, ignoreVersionToResolveSelfDep bool, cloner *rpmrepocloner.RpmRepoCloner) (err error) {
```

Comment:
This name doesn't really mesh with the new behaviour, maybe something like `fixCyclesWithExistingPackages()` or the like would capture it better?